Improve PDF and results download button UI in static embedding

heypoom opened the pull request:
Closes EMB-304

We want to streamline the download button's UI in static embedding, so that:
- PDF export button takes up less space and uses the same design as result downloads.
- PDF export button should adapt to different scenarios (title + filters displayed, only title displayed, only filters displayed, no title and no filters)
- Instead of showing the three-dot menu and having to click "Download results", we should just show the icon button for downloads, which only shows up on hover.
- Tooltips for results and PDF downloads should be consistent

We want to update the download icon button according to [this Figma design file](https://www.figma.com/design/Twp9H8bk1RrhHHgts1RgAu/Disabling-PDF-and-result-downloads-individually-in-static-embeds-and-public-links?node-id=13087-1156&t=QSA8htyHwPj980bO-1).

PS. There isn't as many code changes as it seems, it's mostly Loki reference images that needs to be updated 😄 

### How to verify

Go to static or public embedding in dashboard, and ensure these visual changes are followed. You can refer to the above Figma file as this PR is mostly visual:

- “Export to PDF” in dashboard embeds changes to an icon button
  - It is shown at the top right corner and always visible.
- Download icon for results in both questions and dashboards
   - It moves to the top right on the card and is only shown on hover
- Download button for standalone questions are moved from the footer to the header.
   - The footer should no longer be shown in standalone questions.
- Download buttons all uses the same icon and shows a tooltip on hover.
- See the below section for "Adaptive PDF export button and header" for more on the PDF export position.

You can also look at the Loki diffs to see the changes.

See this [tech doc](https://www.notion.so/metabase/Tech-Disable-PDF-and-result-download-separately-in-static-public-embedding-and-improve-download-UX-1bc69354c901806bb9d6ed29adaf2c4e?pvs=4#1bc69354c90180468477de24d272cc3c) and [testing plan](https://www.notion.so/metabase/Tech-Disable-PDF-and-result-download-separately-in-static-public-embedding-and-improve-download-UX-1bc69354c901806bb9d6ed29adaf2c4e?pvs=4#1bd69354c90180288b6bda7ab2ceab55).

### Adaptive PDF export button / header - scenarios & screenshots

- For dashboard embeds, the header and export icon in the embed frame should change depending on whether title or filter is shown:
  - Title on → show the PDF export icon on the dashboard title
  - Title off, filters on → show the PDF export icon on the right side of the filter
  - Title off, filters off → shrink the height of the header to be compact

There are five dimensions for testing this:
- Is the dashboard title enabled: Y/N
- Are there more than one tabs: Y/N
- Are there at least one dashboard filter: Y/N
- Is the PDF export button shown: Y/N
- Size of the screen/viewport: Desktop or Mobile

Title 🔴, Tab 🔴, Filters 🔴, Download 🟢 -- makes the top header height shorter

![CleanShot 2568-04-10 at 03 32 29@2x](https://github.com/user-attachments/assets/9f88e495-768e-4273-9971-a42656e10cb9)

Title 🟢, Tab 🔴, Filters 🟢, Download 🟢

![CleanShot 2568-04-10 at 03 28 52@2x](https://github.com/user-attachments/assets/1995a13b-19cb-4f23-9000-77eb682e0625)

Title 🔴, Tab 🔴, Filters 🟢, Download 🟢 -- positions the PDF export button in between the filter

![CleanShot 2568-04-10 at 03 30 28@2x](https://github.com/user-attachments/assets/62e6d54d-1781-4c4e-9189-7d7b80eb3cc7)

Title 🔴, Tab 🔴, Filters 🟢, Download 🟢, Mobile Viewport 📱

<img src="https://github.com/user-attachments/assets/1adf90d6-f949-44b0-8291-2fdd39594438" width="300">

Title 🟢, Tab 🔴, Filters 🔴, Download 🟢

![CleanShot 2568-04-10 at 03 31 46@2x](https://github.com/user-attachments/assets/4477bb59-4820-4af9-b771-8ed338b6d31a)

Title 🔴, Tab 🟢, Filters 🟢, Download 🟢 - aligns the PDF export button with the tab bar

![CleanShot 2568-04-10 at 03 36 20@2x](https://github.com/user-attachments/assets/b344f480-bfa5-48b1-a25d-41940d80115a)

Title 🟢, Tab 🟢, Filters 🟢, Download 🟢

![CleanShot 2568-04-10 at 03 37 18@2x](https://github.com/user-attachments/assets/2d5b4928-2f94-4ce3-b2c3-a574bfeca877)

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

heypoom committed:
replace the export button in dashboard header

heypoom committed:
use the same download button style for both

heypoom committed:
add styles for compact download button

heypoom committed:
change the compact export button style

heypoom committed:
add download button logic

heypoom committed:
update data-testid

heypoom committed:
update dashboard tests in public-resource-downloads

heypoom committed:
use getDashboardCard helper instead

heypoom committed:
move question result download to header

heypoom committed:
remove PDF export localized test

heypoom committed:
add PublicOrEmbeddedDashboardView loki diffs

heypoom committed:
use text-primary instead of text-dark for download icons

heypoom commented on .loki/reference/chrome_laptop_App_Embed_PublicOrEmbeddedDashboardView_Card_Visualizations_Dark_Theme.png:
All Loki diffs are because we replaced "Export to PDF" with the download icon - look at the top right.

heypoom committed:
simplify downloadAndAssert embed logic

heypoom commented on e2e/support/helpers/e2e-downloads-helpers.ts:
```diff
@@ -89,10 +93,15 @@ export function downloadAndAssert(
   if (isDashboard) {
-    getDashboardCardMenu().click();
-    cy.findByText("Download results").click();
+    if (isEmbed) {
```
We only show the `download-dashcard-results-button` in the embed UI, we still continue to show the dashcard menu in the main app. The reason is that there is more things you can do in that dashcard menu when in the main app, such as editing the questions.

heypoom commented on e2e/test-component/scenarios/embedding-sdk/locale-on-provider.cy.spec.tsx:
```diff
@@ -82,6 +82,6 @@ describe("scenarios > embedding-sdk > locale set on MetabaseProvider", () => {
     // should not do any request, as `en` doesn't need loading
 
-    getSdkRoot().findByText("Export as PDF").should("exist");
+    getSdkRoot().findByTestId("export-as-pdf-button").should("exist");
```
It's now an icon, so we must look up by test id.

heypoom commented on e2e/test/scenarios/embedding/embed-resource-downloads.cy.spec.ts:
```diff
@@ -53,10 +53,10 @@ H.describeWithSnowplowEE(
 
         // we should not have any dashcard action in a static embedded/embed scenario, so the menu should not be there
-        H.getDashboardCardMenu().should("not.exist");
+        cy.findByTestId("download-dashcard-results-button").should("not.exist");
```
It's now an icon that's part of the dashboard card, as opposed to a menu item.

heypoom committed:
simplify classnames merging

heypoom commented on frontend/src/metabase/dashboard/components/DashboardHeader/buttons/ExportAsPdfButton.module.css:
```diff
@@ -0,0 +1,10 @@
+.CompactExportAsPdfButton {
+  position: absolute;
+  right: 1rem;
+  top: 0.5rem;
+  z-index: 1;
+}
+
+.CompactExportAsPdfButton.ParametersVisibleWithNoTabs {
+  top: 1.75rem;
+}
```
When in compact layout, we use absolute positioning. Also, if there are only parameters (filters) available we align it with the filter box.

heypoom committed:
revert ExportPdfMenuItem

heypoom committed:
cleanup TitleAndDescriptionContainer

heypoom commented on frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.tsx:
```diff
@@ -163,6 +164,8 @@ export function PublicOrEmbeddedDashboardView({
     [externalGetClickActionMode, navigateToNewCardFromDashboard],
   );
 
+  const isCompactHeader = !titled && !hasVisibleParameters && !dashboardHasTabs;
```
Show a very compact header when there's no title, no parameters and no tabs.

![CleanShot 2568-04-11 at 06 15 51@2x](https://github.com/user-attachments/assets/201e41cc-af23-49d8-9f49-da9f7807ef82)

heypoom commented on frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/PublicOrEmbeddedQuestionView.tsx:
```diff
@@ -87,6 +87,7 @@ export function PublicOrEmbeddedQuestionView({
       hide_parameters={hide_parameters}
       theme={theme}
       titled={titled}
+      headerButtons={questionResultDownloadButton}
```
Move the download button from footer to header.

heypoom committed:
revert LegendCaption

npretto reviewed (changes requested):
Code looks good, but I noticed that the download button for (standalone) questions is not visible when the title is hidden, I haven't seen anything in the PR description or linear task that indicates that that's the expected behavior.
<img width="1432" alt="image" src="https://github.com/user-attachments/assets/4df93346-532f-4768-b939-411b92937604" />


Figma also mentions that the download button on cards should only be visible when hovering them: https://www.figma.com/design/Twp9H8bk1RrhHHgts1RgAu/Disabling-PDF-and-result-downloads-individually-in-static-embeds-and-public-links?node-id=13087-1156&t=8UKIpdMXlb51DiOS-4

heypoom committed:
show the header if headerButtons is not null

heypoom committed:
standalone question download results only shows up on card hover

heypoom committed:
always show header if there is a header button

heypoom committed:
add floating headers when title is hidden

heypoom committed:
add comment on floating header buttons

heypoom committed:
use accessible label selectors instead of data-testid

heypoom committed:
replace question-results-download-button testid with accessible label

heypoom committed:
cleanup downloadAndAssert logic

heypoom committed:
update tests

heypoom committed:
support shared selector

heypoom committed:
update unit test for PublicOrEmbeddedDashboardPage.enterprise

heypoom committed:
update unit test for PublicOrEmbeddedDashboardPage.premium

heypoom committed:
update unit test for PublicOrEmbedded

heypoom commented:
> Code looks good, but I noticed that the download button for (standalone) questions is not visible when the title is hidden, I haven't seen anything in the PR description or linear task that indicates that that's the expected behavior.

> Figma also mentions that the download button on [standalone question] cards should only be visible when hovering them.

@npretto Thanks for the catches, I overlooked the standalone questions since I worked on them last 🙏🏻. I fixed these issues in the PRs:

- Updated the selectors to use accessible labels
- When standalone question title is hidden, we show a download button on the top-right corner
- Download button on standalone questions now only shows up once you hovered on the card

heypoom committed:
update localization tests

heypoom committed:
localize aria-label and re-add label tests

heypoom committed:
make aria-label "Download as PDF" for all instances

heypoom committed:
fix question tests on locale hash params

heypoom committed:
fix tests on disabling downloads

heypoom committed:
add realHover case to downloadAndAssert

heypoom committed:
update testid in LegendDescriptionIcon

heypoom committed:
update public-resource-downloads cypress test

heypoom committed:
use visibility checks in tests

heypoom committed:
restore ellipsis button size

heypoom committed:
restore hover parent state to standalone question

heypoom committed:
update localization tests that referenced PDF export button

heypoom committed:
remove eslint-disable directives

heypoom committed:
update ChartSkeleton

heypoom committed:
update ChartSkeleton

heypoom committed:
lift spacing to LegendCaption

npretto reviewed (approved):
Code changes looks good, would love to have a design/product review for this: https://metaboat.slack.com/archives/C063Q3F1HPF/p1744617736696539?thread_ts=1744450714.241579&cid=C063Q3F1HPF before merging this to master

heypoom committed:
use lighter colors for dashcard downloads

heypoom committed:
Merge remote-tracking branch 'origin/master' into emb-304-revise-the-download-buttons-ui-in-static-embedding

heypoom committed:
add new loki snapshots

heypoom committed:
use the auto refresh text for locale tests as pdf export text is gone

heypoom committed:
use the lighter color for all download icons

heypoom committed:
add new loki snapshots

heypoom committed:
Merge branch 'master' into emb-304-revise-the-download-buttons-ui-in-static-embedding

heypoom committed:
update test assertions for locale tests
